Sources/Workspace/UserToolchain.swift

```diff
@@ -40,6 +40,19 @@ public struct UserManifestResources: ManifestResourceProvider {
         self.libDir = libDir
         self.sdkRoot = sdkRoot
     }
+
+    /// Creates the set of manifest resources associated with a `swift` executable.
+    ///
+    /// - Parameters:
+    ///     - swiftExectuable: The absolute path of the associated `swift` executable.
+    public init(swiftExectuable: AbsolutePath) throws {
+        let binDir = swiftExectuable.parentDirectory
+        self.init(
+            swiftCompiler: try UserToolchain.swiftCompilers(
```

Comment:
I don't think we should lookup compilers when one is explicitly being passed in. We could do this if the input was binDir but since this is taking Swift compiler, we should use it as-is.

Reply:
Removed this lookup (and undid the refactor that it necessitated).

(Not pushed yet.)